pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -74,6 +74,7 @@ type ShootSpec struct {
 	// +optional
 	Addons *Addons `json:"addons,omitempty" protobuf:"bytes,1,opt,name=addons"`
 	// CloudProfileName is a name of a CloudProfile object. This field is immutable.
+	// +optional
 	CloudProfileName string `json:"cloudProfileName" protobuf:"bytes,2,opt,name=cloudProfileName"`
```

Comment:
It should be a pointer, to be an optional field.
```suggestion
	CloudProfileName *string `json:"cloudProfileName" protobuf:"bytes,2,opt,name=cloudProfileName"`
```

Reply:
The `CloudProfileName` field is not supposed to be optional yet. It will be migrated by a future PR. I fixed it in https://github.com/gardener/gardener/pull/9440/commits/026d5bdc62ce06a524e6839f6d663a7dcf36a6e6.